### PR TITLE
Ez leka/minikube minishift

### PR DESCRIPTION
**Pull request by ez-leka**
1. Refactored .yaml files to follow Kubernetes best practices
2. Implemented customization based on km tag and URL. The URL will be specifically needed for KM CI 
3. Tested k3s, aks, gek, Minikube. 
4. implemented CI 

Note: minikube was tested with Podman cri-o on Fedora and Ubuntu 20.04 locally. but does not work on GitHub runner or on  EC2 Ubuntu instance, so for now is commented from CI

### CODE REVIEW

File: helpers/minishift-cluster.sh [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+[ "$TRACE" ] && set -x
+
+arg_count=$#
+VER="1.34.3"
```
Comment (gnode1) [inline helpers/minishift-cluster.sh:21]: `?=` ?

File: .github/workflows/ci-workflow.yaml [hunk 2]
```diff
@@ -79,41 +91,114 @@ jobs:
...
+        run: |
+          sudo apt-get update
+          sudo apt-get upgrade -y
+
+          sudo apt-get install -y podman
+
+      - name: Prepare cri-o
+        if: ${{ matrix.runtime == 'cri-o' }}
+        run: |
+          sudo apt-get update
```
Comment (gnode1) [inline .github/workflows/ci-workflow.yaml:134]: Should this be part of minikube provision script?

File: .github/workflows/ci-workflow.yaml [hunk 3]
```diff
@@ -79,41 +91,114 @@ jobs:
...
+          sudo apt-get upgrade -y
+
+          export OS=xUbuntu_20.04
```
Comment (gnode1) [inline .github/workflows/ci-workflow.yaml:137]: x?

File: .github/workflows/ci-workflow.yaml [hunk 4]
```diff
@@ -79,41 +91,114 @@ jobs:
...
+          sudo iptables -P FORWARD ACCEPT
+          sudo iptables -P INPUT ACCEPT
+          sudo iptables -P OUTPUT ACCEPT
+
+          # install cri-o
+          echo "deb https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/$OS/ /" | sudo dd of=/etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list
+          echo "deb http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/$VERSION/$OS/ /" | sudo dd of=/etc/apt/sources.list.d/devel:kubic:libcontainers:stable:cri-o:$VERSION.list
+
+          curl -L https://download.opensuse.org/repositories/devel:kubic:libcontainers:stable:cri-o:$VERSION/$OS/Release.key | sudo apt-key add -
+          curl -L https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/$OS/Release.key | sudo apt-key add -
```
Comment (gnode1) [inline .github/workflows/ci-workflow.yaml:163]: I wasn't sure about the `x` in $OS, so tried these curls manually. And they didn't work, with or without `x`

File: helpers/minikube-cluster.sh [hunk 5]
```diff
@@ -23,8 +23,8 @@ runtime="containerd"
 driver="podman"
 
 print_help() {
-    echo "usage: $0  [OPTIONS | --cleanup]"
-    echo "Creates minicube cluster"
```
Comment (smijar) [inline helpers/minikube-cluster.sh:27]: minikube

File: helpers/minikube-cluster.sh [hunk 6]
```diff
@@ -23,8 +23,8 @@ runtime="containerd"
...
+    echo "usage: $0  [OPTIONS | --cleanup | --stop| --restart]"
+    echo "Creates minikube cluster"
```
Comment (smijar) [inline helpers/minikube-cluster.sh:27]: minikube

File: helpers/minishift-cluster.sh [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+    minishift delete  --clear-cache --force
+    rm -rf ~/.kube/
+    rm -rf minishift-$VER-linux-amd64
+    rm minishift-$VER-linux-amd64.tgz
+    PATH=$(echo "$PATH" | sed -e 's/:\/d\/Programme\/cygwin\/bin$//')
+}
+
+print_help() {
+    echo "usage: $0  [OPTIONS | --cleanup]"
+    echo "Creates minicube cluster"
```
Comment (smijar) [inline helpers/minishift-cluster.sh]: minishift

**Review by smijar — APPROVED** [review]

**Pull request merged**